src/main/scala/com/gossiperl/client/Messaging.scala

```diff
@@ -0,0 +1,73 @@
+package com.gossiperl.client
+
+import java.util
+
+import akka.actor.{Props, ActorLogging, Actor}
+import akka.util.Timeout
+import com.gossiperl.client.actors.ActorRegistry
+import com.gossiperl.client.exceptions.GossiperlClientException
+import com.gossiperl.client.serialization.{Serializer, DeserializeResultForward, DeserializeResultError, DeserializeResultOK}
+import com.gossiperl.client.thrift._
+import com.gossiperl.client.transport.{UdpTransportProtocol, UdpTransport}
+import scala.collection.JavaConverters._
+
+import scala.util.{Failure, Success}
+
+class Messaging(val configuration: OverlayConfiguration) extends ActorRegistry with ActorLogging {
+
+  import scala.concurrent.duration._
+  import scala.concurrent.ExecutionContext.Implicits.global
+
+  context.actorOf(Props( new UdpTransport(configuration) ), name = s"${configuration.overlayName}-transport")
+
+  log.debug(s"Messaging for overlay ${configuration.overlayName} is running.")
+
+  implicit val timeout = Timeout(1 seconds)
+
+  def receive = {
+    case GossiperlProxyProtocol.ShutdownRequest(p) =>
+      log.debug("Received shutdown request.")
+      p.success(self)
+      context.stop( self )
+    case UdpTransportProtocol.IncomingData(result) =>
+      result match {
+        case _:DeserializeResultOK =>
+          result.asInstanceOf[DeserializeResultOK].getDigestType match {
+            case Serializer.DIGEST =>
+              digestAck( result.asInstanceOf[DeserializeResultOK].getDigest.asInstanceOf[ Digest ] )
+            case Serializer.DIGEST_ACK =>
+              !:(s"${configuration.overlayName}-client-state", FSMProtocol.AckReceived)
```

Comment:
Having said that, it would be possible to eliminate the registry and still hide the complexity of `actorSelection()` behind the `!:` operator by using an implicit like (sketch):

    def !:( path: String, message: Any ): Unit = {
      context.system.actorSelection( path ) resolveOne() onComplete {
        case Success( a ) => a ! message
        case Failure(ex) => logger.error(s"Could not resolve actor $path. Message $message will not be delivered.", ex)
      }
    }

It could be clearer, I am not sure about the resolution time of `actorSelection()` though. Something to measure?